lib/procore-sift.rb

```diff
@@ -66,7 +66,16 @@ def sort_fields
 
   class_methods do
     def filter_on(parameter, type:, internal_name: parameter, default: nil, validate: nil, scope_params: [])
-      filters << Filter.new(parameter, type, internal_name, default, validate, scope_params)
+      if type.respond_to?(:key?)
+        unless type.key?(:scope)
+          raise ArgumentError, "filter_on: Only type: :scope can have subtypes. Expecting the form `type: {scope: [type, {param: type}]}`"
+        end
```

Comment:
Maybe move this validation into the Filter class since it has access to `type` and `scope_types` it could do this validation there pretty easily such as in the `validate_scope_types!` method.

Reply:
Good idea. I'll make this change unless we decide on some other `filter_on` interface.